sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/StreamingProducer.cs

```diff
@@ -276,6 +276,49 @@ public async Task EnqueueEventAsync(EventData eventData, SendEventOptions option
             TotalPendingEventCount++;
         }
 
+        /// <summary>
+        /// Enqueues an <see cref="EventData"/> instance to be published to the Event Hub. This method waits until the queue has
+        /// space for the given event, it does not wait until the event is published.
+        /// </summary>
+        /// <param name="eventData">An <see cref="IEnumerable{T}"/> of event data instances to be published.</param>
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the
+        /// operation.</param>
+        /// <param name="options">An optional set of options to consider when publishing this event.</param>
+        /// <remarks>
+        /// <para>
+        /// Upon the first call to queue, the send success and failures handlers cannot be changed.
+        /// </para>
+        ///
+        /// <para>
+        /// The set of events in the <see cref="IEnumerable{T}"/> have no guarantees about which batches each individual <see cref="EventData"/>
+        /// instance get put in. In other words, the group of events listed here are not necessarily batched together.
+        /// </para>
+        ///
+        /// <para>
+        /// When this call returns, the event will have been added to the queue, but may not have been published yet. Publishing the event
+        /// occurs in the background. After attempting to publish, the appropriate handler will be called, either <see cref="SendSucceededAsync"/>
+        /// if the events are sent successfully, or <see cref="SendFailedAsync"/> if the events fail to send.
+        /// </para>
+        /// </remarks>
+        public async Task EnqueueEventAsync(IEnumerable<EventData> eventData, SendEventOptions options = default, CancellationToken cancellationToken = default)
```

Comment:
```suggestion
        public async Task EnqueueEventsAsync(IEnumerable<EventData> eventData, SendEventOptions options = default, CancellationToken cancellationToken = default)
```

We'll probably want to split these into two overloads for the actual arch board design - one with the send options and one without.